### ISSUE TITLE
Allow Certbot to run in parallel

### ISSUE THREAD

**Issue by bmw** [certbot/certbot#4628]
In 0.14.0 we added simple lockfiles to prevent concurrently running Certbot instances from corrupting your files. As you can see in https://github.com/certbot/certbot/pull/4449#issuecomment-299802507, some people had been running Certbot concurrently though. While doing this was unsafe (and undocumented), it'd be nice if things like certificate renewal could be run in parallel to speed up things for large integrations.

Doing this would be a significant increase in complexity to Certbot's locking mechanism though.

**Comment by vrubiella** [comment]
Yes please!, no-lock option or lock-domain option.

**Comment by mattlqx** [comment]
I'd really love to see the ability for Certbot to spawn multiple instances of the `--manual-auth-hook`/`--manual-cleanup-hook` scripts so that multiple DNS challenges in the same certificate request (ala SANs) can be processed in parallel. This would speed up individual requests significantly if you're using SANs and are using a DNS provider that has some settling time on each transaction.

**Comment by bmw** [comment]
While I certainly see the benefits of this, to be honest, I really doubt the small Certbot team will ever have time to get to this so I'm closing this issue and marking it as wontfix.

**Comment by PELock** [comment]
Well it's a good reason to allow it to run in parallel.

I have 3 domains expiring at the same time, I was out of home and it expired, all 3 of them.

- I had to run certbot-auto with DNS validation on each of them
- It failed 3 times (like it always does, because since the DNS gets refreshed, certbot-auto fails to read it and generates a new code each time, it's been like that on certbot-auto v1.0.x for over a year)
- I have to wait for like 45 minutes for each domain to get a proper cert
- So it's 3x 45 min when I need it the most!

**Comment by bmw** [comment]
@PELock, if Certbot failing the challenge three times is a different issue than https://github.com/certbot/certbot/issues/7679, I'd encourage you to make a post at https://community.letsencrypt.org describing the problem. If it's a problem in Certbot, they'll probably send you back here to create an issue, but if it's a problem with your setup, they should be able to help you through it.

I'd also encourage you to automate your renewal by following the instructions at https://certbot.eff.org/lets-encrypt/pip-other. If you are using the manual plugin, you'll either need to switch to another plugin or write scripts to automate performing the challenges as described at https://certbot.eff.org/docs/using.html#hooks. If you need any help with this, I'd also recommend making a post at https://community.letsencrypt.org.
